dali/base/dadfs.cpp

```diff
@@ -3295,9 +3295,21 @@ class CDistributedFileBase : implements INTERFACE, public CInterface
     virtual void setAccessedTime(const CDateTime &dt) = 0;                      // set date and time last accessed
     virtual bool isExternal() const { return external; }
 
-    virtual int getExpire()
+    virtual int getExpire(StringBuffer *expirationDate)
     {
-        return queryAttributes().getPropInt("@expireDays", -1);
+        int expireDays = queryAttributes().getPropInt("@expireDays", -1);
+        if (!expirationDate || (expireDays == -1))
+            return expireDays;
+
+        const char *lastAccessed = queryAttributes().queryProp("@accessed");
+        if (isEmptyString(lastAccessed))
+            return expireDays;
```

Comment:
Shouldn't this set expirationDate to something?

Reply:
To my understand, since the @accessed is not set, the file will not be expired until the @accessed is set. I am not sure we should set expirationDate to "x days after the file is acccessed".

Reply:
@jake, do you know?

Reply:
I think it's okay as it is. The "accessed" attribute is set by the engines only, so could be missing. If it is, this function can't calculate the expiration date and leaves that param unfilled.

It does beg the question of whether "accessed" should ever be missing if expireDays is set.
Perhaps if unset and setExpire is called, it should also set \@accessed at that point - but that's a separate issue.

NB: files with \@expireDays but without \@accessed will not be deleted (by Sasha).